drivers/python/rethinkdb/_export.py

```diff
@@ -122,6 +124,11 @@ def parse_options():
     else:
         res["fields"] = options.fields.split(",")
 
+    if options.delimiter is not None and options.format != "csv":
+        raise RuntimeError("Error: You can only specify a delimiter when writing in CSV format.")
+
+    res["delimiter"] = ',' if options.delimiter is None else options.delimiter
```

Comment:
Similar code in `_import.py` verifies that the `delimiter` option is at most one character, and also allows for "\t" to be specified.